.github/workflows/pr-commands-closed.yml

```
@@ -0,0 +1,17 @@
+name: Run when PRs are closed
+on:
+  pull_request:
+    types:
+      - closed      
+
+jobs:
+  close_job:
+    # this job will only run if the PR has been closed without being merged
+    if: github.event.pull_request.merged == false
+    runs-on: ubuntu-latest
+    steps:
+    - run: |
+        echo PR #${{ github.event.number }} has been closed without being merged, removing all milestones.
```

There could only be one milestone at maximum.

```suggestion
        echo PR #${{ github.event.number }} has been closed without being merged, removing milestone.
```